drf_spectacular/plumbing.py

```diff
@@ -1126,6 +1126,23 @@ def _get_type_hint_origin(hint):
         return origin, args
 
 
+def _resolve_typeddict(hint):
+    """resolve required fields for TypedDicts if on 3.9 or above"""
+
+    required = None
+
+    if sys.version_info >= (3, 9):
```

Comment:
i'm going to do some quick testing sometime this week to see if it just works with the typing_extension TypedDict as well (what we're using since we're not on 3.9 yet), in which case we can simply check the hint for the `__required_keys__` attribute.